nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -547,21 +560,32 @@ in {
             ''}
 
             ln -sf ${cfg.package}/apps ${cfg.home}/
+            rm -rf ${cfg.home}/nix-apps
+            mkdir -p ${cfg.home}/nix-apps
+
+            #Install extra apps
+            '' +
+            builtins.concatStringsSep "\n" (
+              lib.mapAttrsToList
+                (name: target:
+                  "ln -sf   ${target} ${cfg.home}/nix-apps/${name}")
+                cfg.extraApps) +
```

Comment:
I think it may be nicer, easier to read and perhaps less error-prone to use `buildEnv` to create a store-path with all apps inside. This can be replaced atomically then in `nextcloud-setup` by replacing the symlink on `${cfg.home}/nix-apps` only.

Reply:
I did it with [`linkFarm`](https://github.com/NixOS/nixpkgs/blob/4be93515614857491efe5ec00eba8105b893cce3/pkgs/build-support/trivial-builders.nix#L360) instead. `buildEnv` joined the contents of the apps in one folder